### PR TITLE
Compliance opt-in changes

### PR DESCRIPTION
**Pull request by vara2504**
https://tigera.atlassian.net/browse/EV-4834

From 3.20, compliance components will not be deployed by default. User would be enabling compliance feature to get the resources deployed

Product Version(s):

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico-cloud/compliance/enable-compliance.mdx
Comment (dimitri-nicolo) [inline calico-cloud/compliance/enable-compliance.mdx:1]: Change all references of `compliance` to a lower case `c`, for consistency.

File: calico-enterprise/compliance/enable-compliance.mdx
Comment (dimitri-nicolo) [inline calico-enterprise/compliance/enable-compliance.mdx:1]: Change all references of compliance to a lower case c, for consistency.

File: calico-cloud/compliance/enable-compliance.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  name: tigera-secure
+EOF
+```
+
+### Enable Compliance using Manager UI
+
+On the Manager UI, click **Compliance Reports**, **Configure Compliance Reports**.
+
+:::note
+Enabling packet capture might take a few seconds to create the compliance deployment, daemonset and other dependent resources.
```
Comment (dimitri-nicolo) [inline calico-cloud/compliance/enable-compliance.mdx]: Should it be `Enabling compliance` ?

File: calico-enterprise/compliance/enable-compliance.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  name: tigera-secure
+EOF
+```
+
+### Enable Compliance using Manager UI
+
+On the Manager UI, click **Compliance Reports**, **Configure Compliance Reports**.
+
+:::note
+Enabling packet capture might take a few seconds to create the compliance deployment, daemonset and other dependent resources.
```
Comment (dimitri-nicolo) [inline calico-enterprise/compliance/enable-compliance.mdx]: Should it be `Enabling compliance` ?

File: calico-enterprise/compliance/enable-compliance.mdx
Comment (dimitri-nicolo) [inline calico-enterprise/compliance/enable-compliance.mdx:1]: Are there any prerequisites of note we want to highlight?

File: calico-cloud/compliance/enable-compliance.mdx
Comment (dimitri-nicolo) [inline calico-cloud/compliance/enable-compliance.mdx:1]: Are there any prerequisites of note we want to highlight?

**Review by dimitri-nicolo — CHANGES_REQUESTED** [review]

File: calico-enterprise/compliance/enable-compliance.mdx
Comment (vara2504) [reply calico-enterprise/compliance/enable-compliance.mdx:1]: I don't thing there would be any prerequisites for enabling compliance.
However, there be prerequisites to enable compliance before other configuring compliance features hence made changes for those.

File: calico-cloud/compliance/compliance-reports-cis.mdx [hunk 3]
```diff
@@ -26,6 +26,14 @@ We recommend that you review the CIS benchmark best practices for securing clust
 
 ## Before you begin
 
+**Required**
+
+- Ensure compliance is enabled in the cluster.  If not, [Enable Compliance](../compliance/enable-compliance)
+
+```bash
+kubectl get compliance tigera-secure
+```
```
Comment (ctauchen) [inline calico-cloud/compliance/compliance-reports-cis.mdx]: What is this for?

File: calico-cloud/compliance/compliance-reports-cis.mdx [hunk 4]
```diff
@@ -26,6 +26,14 @@ We recommend that you review the CIS benchmark best practices for securing clust
 
 ## Before you begin
 
+**Required**
+
+- Ensure compliance is enabled in the cluster.  If not, [Enable Compliance](../compliance/enable-compliance)
```
Comment (ctauchen) [inline calico-cloud/compliance/compliance-reports-cis.mdx]:
```suggestion
* You enabled [compliance reporting](../compliance/enable-compliance).
```

File: calico-cloud/compliance/enable-compliance.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+description: Enable compliance to configure reports to assess compliance for all assets in a Kubernetes cluster.
+---
+
+# Enable Compliance
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: This title doesn't sound quite right to me, but maybe I'm just not familiar with the jargon. "Compliance" sounds like a large, abstract concept. The function of this set of features is to enable reporting that lets you figure out whether your system complies with various standards, right? 

If I'm not off here, we should go with a title along the lines of "Enable compliance reporting".

File: calico-cloud/compliance/enable-compliance.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Big picture
+
+Enabling compliance improves the cluster's compliance posture. It involves generating compliance reports for Kubernetes clusters based on archived flow and audit logs for Calico Enterprise and Kubernetes resources. The process includes components for snapshotting configurations, generating reports, managing jobs, providing APIs with RBAC, and benchmarking security.
+
+## Value
+
+The compliance system consists of several key components that work together to ensure comprehensive compliance monitoring and reporting:
+
+ - **compliance-snapshotter:** Lists required configurations and pushes snapshots to Elasticsearch, providing visibility into configuration changes.
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: These look like component names. Better to have these formatted in `code font`.

File: calico-cloud/compliance/enable-compliance.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+EOF
+```
+
+### Enable Compliance using Manager UI
+
+On the Manager UI, click **Compliance Reports**, **Configure Compliance Reports**.
+
+:::note
+Enabling compliance might take a few seconds to create the compliance deployment, daemonset and other dependent resources.
+:::
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: Unless this is a truly long time, remove this note. It's fair to expect users to understand it's not instant. And I would hope there's some kind of status indicator in the UI.

File: calico-cloud/compliance/enable-compliance.mdx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+ - **compliance-snapshotter:** Lists required configurations and pushes snapshots to Elasticsearch, providing visibility into configuration changes.
+ - **compliance-reporter:** Generates reports by analyzing configuration history, determining configuration evolution and identifying "worst-case outliers."
+ - **compliance-controller:** Manages the creation, deletion, and monitoring of report generation jobs.
+ - **compliance-server:** Offers API for report management and enforces RBAC.
+ - **compliance-benchmarker:** Runs CIS Kubernetes Benchmark checks on each node to ensure secure deployment.
+
+### Enable Compliance using kubectl
+
+Create a compliance custom resource, named `tigera-secure`, in the cluster.
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: If a procedure has only one step, format as a bulleted list. So:

`* Create a compliance ...`

File: calico-cloud/compliance/enable-compliance.mdx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: operator.tigera.io/v1
+kind: Compliance
+metadata:
+  name: tigera-secure
+EOF
+```
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: And don't forget to indent this block to match the instruction.

File: calico-cloud/compliance/enable-compliance.mdx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+### Enable Compliance using Manager UI
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]:
```suggestion
### Enable compliance (reporting?) using Manager UI
```

File: calico-cloud/compliance/enable-compliance.mdx [hunk 11]
```diff
@@ -0,0 +1,42 @@
+---
+description: Enable compliance to configure reports to assess compliance for all assets in a Kubernetes cluster.
+---
+
+# Enable Compliance
+
+## Big picture
+
+Enabling compliance improves the cluster's compliance posture. It involves generating compliance reports for Kubernetes clusters based on archived flow and audit logs for Calico Enterprise and Kubernetes resources. The process includes components for snapshotting configurations, generating reports, managing jobs, providing APIs with RBAC, and benchmarking security.
```
Comment (ctauchen) [inline calico-cloud/compliance/enable-compliance.mdx]: I'm not sure that first statement is true. Doesn't this just enable reporting?

File: calico-cloud/compliance/enable-compliance.mdx (hunk 5 above)
Comment (ctauchen) [reply calico-cloud/compliance/enable-compliance.mdx]: We should change instances of 'compliance' to 'compliance reporting' throughout, where applicable.

**Review by ctauchen — COMMENTED** [review]
A few comments for you @vara2504. Thanks for this!

File: calico-cloud/compliance/compliance-reports-cis.mdx (hunk 3 above)
Comment (vara2504) [reply calico-cloud/compliance/compliance-reports-cis.mdx]: this is the command to verify compliance is enabled in cluster

File: calico-cloud/compliance/enable-compliance.mdx (hunk 5 above)
Comment (vara2504) [reply calico-cloud/compliance/enable-compliance.mdx]: compliance include compliance reports, compliance snapshot and compliance benchmark. How able using "compliance features" instead @ctauchen

File: calico-cloud/compliance/enable-compliance.mdx (hunk 11 above)
Comment (vara2504) [reply calico-cloud/compliance/enable-compliance.mdx]: compliance include compliance reports, compliance snapshot and compliance benchmark. How able using "compliance features" instead @ctauchen

File: calico-cloud/compliance/enable-compliance.mdx (hunk 5 above)
Comment (ctauchen) [reply calico-cloud/compliance/enable-compliance.mdx]: I get that there's a specific component that deals with reports, while others do other things. But I still think the general idea here is all in service of the compliance reports, no?

And in any case, we should continue to match the UI, which is clearly about enabling compliance reports, not 'compliance' or 'compliance features'. 

We can open the conversation wider if necessary, but I'm pretty confident we should just change these to 'compliance reports' throughout.

File: calico-cloud/compliance/compliance-reports-cis.mdx [hunk 12]
```diff
@@ -26,6 +26,14 @@ We recommend that you review the CIS benchmark best practices for securing clust
 
 ## Before you begin
 
+**Required**
+
+* You enabled [Enable Compliance](../compliance/enable-compliance)
+
+    ```bash
+    kubectl get compliance tigera-secure
+    ```
```
Comment (ctauchen) [inline calico-cloud/compliance/compliance-reports-cis.mdx]: Remove. Users can visit the doc to figure out details about the requirement.

**Comment by vara2504** [comment]
@ctauchen Addressed all the review comments

**Review by dimitri-nicolo — APPROVED** [review]
lgtm

**Comment by ctauchen** [comment]
> @ctauchen Addressed all the review comments

@vara2504 , there's still the changes for complance --> compliance reports. This applies to all titles and other similar usage.

File: calico-cloud/compliance/enable-compliance.mdx (hunk 5 above)
Comment (ctauchen) [reply calico-cloud/compliance/enable-compliance.mdx]: FYI @vara2504

**Pull request merged**